src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,173 @@
+        setElevatorMotorsVoltage(targetState);
+    }
+
+    private void setAngleMotorsVoltage(TrapezoidProfile.State targetState){
+        masterAngleMotor.setVoltage(calculateAngleMotorOutput(targetState));
+        followerAngleMotor.setVoltage(calculateAngleMotorOutput(targetState));
+    }
+
+    private void setElevatorMotorsVoltage(TrapezoidProfile.State targetState){
+        masterElevatorMotor.setVoltage(calculateElevatorMotorOutput(targetState));
+        followerElevatorMotor.setVoltage(calculateElevatorMotorOutput(targetState));
+    }
+
+    private double calculateAngleMotorOutput(TrapezoidProfile.State targetState){
+        double pidOutput = ArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAngleMotorPosition().getDegrees(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
+        );
+
+        return pidOutput + feedForward;
+    }
+
+    private double calculateElevatorMotorOutput(TrapezoidProfile.State targetState){
+        double pidOutput = ArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorMotorPositionRevolutions(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ELEVATOR_FEEDFORWARD.calculate(
+                Units.rotationsToRadians(targetState.position),
+                Units.rotationsToRadians(targetState.velocity)
+        );
+
+        return pidOutput + feedForward;
+    }
+
+    private void generateAngleMotorProfile(Rotation2d targetAngle) {
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAngleMotorPosition().getDegrees(), getAngleMotorVelocityDegreesPerSecond())
+        );
+
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(double targetElevatorPosition) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                ArmConstants.ELEVATOR_CONSTRAINTS,
+                new TrapezoidProfile.State(targetElevatorPosition, 0),
+                new TrapezoidProfile.State(getElevatorMotorPositionRevolutions(), getElevatorVelocityRevolutionsPerSecond())
+        );
+
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private double getAngleMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleMotorProfileGenerationTime;
+    }
+
+    private double getElevatorMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorMotorProfileGenerationTime;
+    }
+
+    private Rotation2d getAngleMotorPosition() {
+        return Rotation2d.fromRotations(angleEncoder.getPosition().getValue());
```

Comment:
Use Status Signals the way I taught today

Reply:
I added the status signals for the constants to the angle, but it didn't work with the elevator

Reply:
The elevator uses a TalonSRX. As mentioned in the presentation, TalonSRX is phoenix 5. Status signals are phoenix 6

Reply:
and I don't know what to do with it here